File: Leetcode 3901-4000/3921/3921.py
# Leetcode 3921: Score Validator
# https://leetcode.com/problems/score-validator/
# Solved on 22nd of May, 2026
class Solution:
    def scoreValidator(self, events: list[str]) -> list[int]:
        """
        Validates and calculates the final score and event counter based on a list of events.

        :param events: A list of strings representing game events.
        :return: A list of two integers: [final score, event counter].
        """
        currentScore = 0
        currentCounter = 0

        for currentEvent in events:
            if currentEvent == "W":
                currentCounter += 1
                if currentCounter == 10:
                    break
            elif currentEvent in ("WD", "NB"):
                currentScore += 1
            else:
                currentScore += int(currentEvent)

        return [currentScore, currentCounter]

File: Leetcode 3901-4000/3921/3921.java
// Leetcode 3921: Score validator
// https://leetcode.com/probblems/score-validator/
// Solved on 22nd of May, 2026
class Solution {
    /**
     * Validates and calculates the total score and wicket count based on a series of events.
     *
     * @param events An array of strings representing match events (e.g., "W" for wicket, "WD"/"NB" for extras, or digit strings for runs).
     * @return An integer array where the first element is the total score and the second is the total wickets.
     */
    public int[] scoreValidator(String[] events) {
        int score = 0;
        int counter = 0;
        for (String event : events) {
            if (event.equals("W")) {
                counter++;
            } else if (event.equals("WD") || event.equals("NB")) {
                score++;
            } else {
                score += event.charAt(0) - '0';
            }
            if (counter == 10) {
                break;
            }
        }
        return new int[]{score, counter};
    }
}